tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/validation/TclParameterUsageValidatorTest.xtend

```diff
@@ -0,0 +1,292 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.tcl.dsl.validation
+
+import com.google.inject.Provider
+import javax.inject.Inject
+import org.eclipse.xtext.junit4.util.ParseHelper
+import org.eclipse.xtext.junit4.validation.ValidationTestHelper
+import org.eclipse.xtext.resource.XtextResourceSet
+import org.junit.Before
+import org.junit.Test
+import org.testeditor.aml.AmlModel
+import org.testeditor.aml.dsl.AmlStandaloneSetup
+import org.testeditor.tcl.dsl.tests.parser.AbstractParserTest
+import org.testeditor.tml.TmlModel
+import org.testeditor.tml.dsl.TmlStandaloneSetup
+
+import static org.testeditor.tml.TmlPackage.Literals.*
+
+class TclParameterUsageValidatorTest extends AbstractParserTest {
```

Comment:
Can we somehow extract something here? I like the class but 300 lines for a couple of assertions seems a bit much.